lib/kino/proxy.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Kino.Proxy do
+  @moduledoc """
+  Functionality for handling proxy requests forwarded from Livebook.
+
+  Livebook proxies requests at the following paths:
+
+    * `/sessions/:id/proxy/*path` - a notebook sessions
+
+    * `/apps/:slug/:session_id/proxy/*path` - a specific app session
+
+    * `/apps/:slug/proxy/*path` - generic app path, only supported for
+      single-session apps
+
+  You can define a custom listener to handle requests at these paths.
+  The listener receives a `Plug.Conn` and should use the `Plug` API
+  to send the response, for example:
+
+      Kino.Proxy.listen(fn conn ->
+        Plug.Conn.send_resp(conn, 200, "hello")
+      end
+
+  The following `Plug.Conn` fields are set:
+
+    * `:host`
+    * `:method`
+    * `:owner`
+    * `:port`
+    * `:remote_ip`
+    * `:query_string`
+    * `:path_info`
+    * `:scheme`
+    * `:script_name`
+    * `:req_headers`
```

Comment:
These are all "request fields", so perhaps we should say that instead?

Reply:
I don't think we should say which fields are set. For all intents and purpose, it should work as a regular webserver, and that's what a regular webserver would set.